repokid/utils/roledata.py

```diff
@@ -139,39 +142,36 @@ def update_aardvark_data(account, aardvark_data):
             LOGGER.error('Dynamo table error: {}'.format(e))
 
 
-def update_role_data(role_dict):
+def update_role_data(role, current_policy):
     """Given role data either add it to the datastore, add a revision of the policies, or refresh updated time"""
     from repokid.repokid import LOGGER
-    # need to convert to (stupid) DynamoDB empty string form
-    if 'policies' in role_dict:
-        role_dict['policies'] = _empty_string_to_dynamo_replace(role_dict['policies'])
 
     # policy_entry: source, discovered, policy
-    stored_role = _get_role_data(role_dict['RoleId'])
+    stored_role = _get_role_data(role.role_id, fields=['Policies'])
+
     if stored_role:
         # is the policy list the same as the last we had?
-        if not role_dict['policies'] == stored_role['Policies'][-1]['Policy']:
-            add_new_policy_version(role_dict, 'Scan')
-            LOGGER.info('{} has different inline policies than last time, adding to role store'.format(
-                        role_dict['Arn']))
-        _refresh_updated_time(role_dict['RoleId'])
+        if not current_policy == _empty_string_from_dynamo_replace(stored_role['Policies'][-1]['Policy']):
```

Comment:
awkward way to phrase this.

Should it be `if current_policy != ...` instead of `if not current_policy == ...` ?